Source/Kernel/Grains/Observation/Observer.cs

```diff
@@ -112,15 +113,26 @@ public async Task Subscribe<TObserverSubscriber>(
 
         logger.Subscribing();
 
-        State = State with { Type = type, EventTypes = eventTypes };
+        var isReplayable = true;
+        object? actualSubscriberArgs = subscriberArgs;
+
+        // Check if subscriberArgs contains IsReplayable information
+        if (subscriberArgs is ReactorObserverSubscriptionArgs reactorArgs)
```

Comment:
This is a violation of the open closed principle. The Observer in the abstract should not need to know about specific implementing types. Lets move the isReplayable for now onto the `.Subscribe()` method for the observer.